Fix advanced settings

dotalchemy opened the pull request:
Fixed a thing introduced by a change made to the 'community.vmware' collection where `customsettings` was replaced with `advanced_settings` in a recent release of the collection.

This caused the role to fail with `Unable to find custom value key disk.EnableUUID` 

See [Github Issue #179](https://github.com/ansible-collections/community.vmware/pull/179)
See [vmware_guest_module](https://docs.ansible.com/ansible/latest/collections/community/vmware/vmware_guest_module.html)

Tested on whatever version of Ansible is in the very latest AWX Operator build, but there shouldn't be a version dependency change as it's just a string edit in the yaml to match what `community.vmware` is expecting.

dotalchemy committed:
Update main.yml

customvalues --> advanced_settings

dotalchemy committed:
Update README.md

Updated for `advanced_settings`

albers commented:
Thanks very much for your contribution.
I will need some time to review it because I first need to set up suitable testing environments.

albers commented on README.md:
```diff
@@ -18,7 +18,7 @@ This role adds support for these features.
 - Adds one or more ssh public keys and/or a password for the default user "ubuntu" so that Ansible can connect to the new machine.
 - Optionally adjusts the hardware, e.g. number of CPUs or memory, see [vmware_guest](https://docs.ansible.com/ansible/latest/modules/vmware_guest_module.html#parameters) for possible customizations.
-- Optionally sets VM notes and/or VM [customvalues](https://stackoverflow.com/a/57976458/2402612).
+- Optionally sets VM notes and/or VM [Configuration Parameters](https://docs.vmware.com/en/VMware-vSphere/6.7/com.vmware.vsphere.html.hostclient.doc/GUID-8C639077-FF16-4D5D-9A7A-E16902CE00C2.html).
```
Please change the link to use the 7.0 documentation.

albers commented on tasks/main.yml:
```diff
@@ -38,7 +38,7 @@
     name: "{{ vm_guestname }}"
     annotation: "{{ annotation | default(omit) }}"
     hardware: "{{ hardware | default(omit) }}"
-    customvalues: "{{ customvalues | default(omit) }}"
```
An alternative to replacing `customvalues` with `advanced_settings` could be to add the new option and leave the `customvalues` in place.  After all, this option still exists, and maybe there are other use cases of `customvalues` than the one documented in the module.

We could increase portability this way. WDYT?

dotalchemy replied:
Yep, you're 100% on this - will make the changes accordingly!

dotalchemy replied:
added `customvalues` back in and tested - they are set correctly according to the definition in the playbook, though custom value keys won't be created if they don't exist - they must exist in vCenter first in order to be assigned a value

albers reviewed (changes requested):
I did some research:

`advanced_settings` was added to *vmware_guest* in the [1.8.0 release](https://github.com/ansible-collections/community.vmware/blob/main/CHANGELOG.rst#v1-8-0) of the _community.vmware collection_.
This release is part of the Ansible community package 3.2.0, which is based on ansible-core 2.10.7 (see [build data](https://github.com/ansible-community/ansible-build-data/blob/main/3/CHANGELOG-v3.rst#ansible-base-4)).

Because this change is a breaking one, `min_ansible_version` will have to be changed to 2.10.7 [here](https://github.com/hamburger-software/ansible-role-vmware_ubuntu_cloud_image/blob/97d8908ec75544781ec87a55c2b8999315da4a8b/meta/main.yml#L7).

Please also change the [requirements section](https://github.com/hamburger-software/ansible-role-vmware_ubuntu_cloud_image/blob/97d8908ec75544781ec87a55c2b8999315da4a8b/README.md?plain=1#L45) to
"The minimum *community.vmware collection* version is 1.8.0, which is part of the Ansible community package 3.2.0."

dotalchemy committed:
Added customvalues back with notes in README.md for usage
Updated documentation to be vSphere 7.0
Updated documentation to reflect new minimum requirements
Updated galaxy metadata for new minimum Ansible version

dotalchemy reviewed (commented):
Please review and let me know if there are any changes you'd like to see!

albers reviewed (approved):
Perfect, thank you very much for the contribution.

albers commented:
@dotalchemy released [v6.0.0](https://github.com/hamburger-software/ansible-role-vmware_ubuntu_cloud_image/releases/tag/v6.0.0)
